x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/RemoteAccessAuthentication.java

```diff
@@ -56,6 +56,9 @@ public void writeToContext(final ThreadContext ctx) throws IOException {
     }
 
     public static RemoteAccessAuthentication readFromContext(final ThreadContext ctx) throws IOException {
+        if (ctx.getHeader(REMOTE_ACCESS_AUTHENTICATION_HEADER_KEY) == null) {
```

Comment:
This wasn't here before but it belongs here to avoid an uninformative NPE below.

Reply:
Nit: can we extract `ctx.getHeader(REMOTE_ACCESS_AUTHENTICATION_HEADER_KEY)` to a variable so it is reused below?